python-backend/app/api/mines/variances/models/variance.py

```diff
@@ -68,22 +73,28 @@ def create(
 
     @classmethod
     def find_by_mine_guid(cls, mine_guid):
-        try:
-            uuid.UUID(str(mine_guid), version=4)
-            return cls.query.filter_by(mine_guid=mine_guid).all()
-        except ValueError:
-            return None
+        cls.validate_guid(mine_guid, INVALID_MINE_GUID)
```

Comment:
This actually changes the functionality of this method, throwing an error if the guid is invalid rather than returning None. Are we good with this?

Reply:
That was an intentional change. I guess there could be an argument to keep the `None` check + error raising at every invocation of this lookup, but I think it's valuable to catch the invalid GUID in the model and avoid duplicating that `None` check everywhere.

Not a big deal for me to revert this change if we prefer the original way that I was doing this.

If we wanted to suppress the error for some reason, we could always try/except at the callsite. It seems to me that that would be the odd use case, and we'd usually want to throw for the invalid GUID.

This does raise the point that I should remove any existing `None` checks.